tests/test_autocommit.py

```diff
@@ -0,0 +1,6 @@
+import sys, os
+
+def test():
+    "Verify autocommit just before program exits."
+    assert(0 == os.system('%s tests/autocommit.py' % sys.executable))
+    assert(0 == os.system('%s tests/autocommit2.py' % sys.executable))
```

Comment:
Why do you need to run tests/autocommit2.py in a separate subprocess? This is just reading code. Shouldn't you be able to do the same thing directly, inside this `test` function?

e.g. something like:

```python
d = sqlitedict.SqliteDict('tests/db/autocommit.sqlite', autocommit=True)
expected = {i: i for i in range(1000)}
actual = {key: value for (key, value) in d.items()}
assert expected == actual
```

Reply:
Also, please delete the temporary file after you're done. Unit tests should clean up after themselves.

Reply:
Thank you for reviewing my changes.  I will respond to each of your questions here:
1) Temporary files:
I was following the pattern used by the other tests in this project.  None of them delete their temporary files when they are finished.  The `make test-all` target cleans up before running the tests.

2) Separate sub-process:
This is required as per my comment on Aug 11: https://github.com/RaRe-Technologies/sqlitedict/pull/94#issuecomment-520232139.  The source of the bug is actually a race between when threads exit and if you do not exit the process before trying to read the database, then you will not trigger the bug.  If you run the code you suggested, it does not trigger a failure even without my patch.

3) The extra sqlitedict.py file (symlink):
That was also explained in my Aug 11 comment.  Because I'm running a separate script, and that script resides in the "test" directory, the python path does NOT pick up the local sqlitedict module, instead it picks up the system version of the library.  I could have avoided that by modifying `sys.path` inside my test scripts, but I figured it was easier to include a symlink.  If you'd like me to modify the sys.path instead, let me know and I'd be happy to change it.

4) assert syntax:
Sure, I'll remove the parentheses.

Reply:
1. OK
2. Please see below
3. Specify the PYTHONPATH when calling your scripts (e.g. https://docs.python.org/2/library/subprocess.html#subprocess.Popen)
4. OK

---

So, that leaves the second point. I've read through your response several times, and I still don't understand why you need a separate script for reading the DB.

> The source of the bug is actually a race between when threads exit and if you do not exit the process before trying to read the database, then you will not trigger the bug.

The race only happens inside the _writing_ script, right? After the writing script has finished, then the problem has already occurred. So, where you _read_ the database shouldn't really matter (either as a separate script, or inside the actual test).

In case my original point wasn't clear enough, I'm suggesting you do this:

```python
def test():
    "Verify autocommit just before program exits."
    assert(0 == os.system('%s tests/autocommit.py' % sys.executable))

    # The above script relies on the autocommit feature working correctly.
    # Now, let's check if it actually worked.
    # Totally untested code below.
    d = sqlitedict.SqliteDict('tests/db/autocommit.sqlite', autocommit=True)
    expected = {i: i for i in range(1000)}
    actual = {key: value for (key, value) in d.items()}
    assert expected == actual
```

Does that make sense, or did I miss something?

Reply:
BTW, do you need `autocommit=True` when reading the dictionary? I suspect that you probably don't.

Reply:
Oh, I understand what you mean now about doing it as a single test script.  Good idea.  I will make that change.

Reply:
All changes complete.